CMakeLists.txt

```diff
@@ -1,14 +1,30 @@
-# This file is intended for use by library developers/maintainers.
-# It declares targets for unit tests and benchmarks.
-# If you only want to use the library consume lib/CMakeLists.txt instead.
+# This file is intended for use by library developers/maintainers, as well as users and package managers.
+# It declares targets for unit tests and benchmarks and exposes options.
+# This CMake project supports being embedded into other projects using add_subdirectory.
+# This CMake project also support being found by find_package(vmcontainer).
+# Also supports being used with FetchContent
 
 cmake_minimum_required(VERSION 3.12)
 
-project(vmcontainer)
+# TODO: To replace by PROJECT_IS_TOP_LEVEL when requiring CMake 3.21
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    set(VMCONTAINER_TOPLEVEL ON)
+else()
+    set(VMCONTAINER_TOPLEVEL OFF)
+endif()
```

Comment:
Instead of this check here I think what would be better is adding a `VMCONTAINER_DEV_MODE` option which is set to `ON` in the CMake `base` preset (and `OFF` in this file by default). In the future I want to use that flag to also control developer-specific build flags like warnings, sanitizers, etc.

Though I wanted to avoid all these options I guess there's no way around it, so if you do this one change then we're good to go.

Reply:
Very good idea! Also much less fragile than detecting toplevel project.

Reply:
Sorry for the delay. I changed to the model you proposed. I changed the preset file, but I'm not sure if the change is good, I never worked with presets before, so tell me if there's something you would change.